08_education/preschool_county.qmd

```diff
@@ -0,0 +1,766 @@
+
+Remove records from counties below the size quality cutoff. 
+
+```{r}
+#| label: subset-race-ethnicity
+
+acs_race_ethnicity <- acs_race_ethnicity %>% 
+  filter(effective_sample >= 30)
+
+```
+
+### Calculate metric
+
+```{r}
+#| label: calc-race-ethnicity
+
+results_race_ethnicity <- acs_race_ethnicity %>%
+  group_split(statefip) %>% 
+  map_dfr(~ calc_survey_mean_subgroup(.data = .x,
+                                      .geo_level = county,
+                                      .subgroup = subgroup_race))
+```
+
+```{r}
+#| label: finalize-race-ethnicity
+
+results_race_ethnicity <- finalize_metric(results_race_ethnicity)
+
+```
+
+```{r}
+rm(acs_race_ethnicity)
+```
+
+## Add Suppressed Rows to Data
+
+To create the metric we suppressed counties that did not meet the threshold for sample size but we need to include these in the final data. Create an all version of the data that includes counties that were suppressed prior to the survey_mean calculation. 
+
+Create expanded version for race-ethnicity.
+```{r}
+results_race_ethnicity_expanded <- acs_crosswalked %>% 
+  filter(sample_structure == "5year") %>% 
+  mutate(geoid = paste0(statefip, county)) %>% 
+  group_by(year, geoid, subgroup_race) %>% 
+  count() %>% 
+  expand_grid() %>%  
+   mutate(statefip = str_sub(geoid, 1, 2),
+         county = str_sub(geoid, 3, 5)) %>% 
+  left_join(results_race_ethnicity, by = c("year", "statefip", "county", "subgroup_race")) %>% 
+  mutate(subgroup_type = "race-ethnicity") %>% 
+  rename(subgroup = subgroup_race)
+```
+
+## Data Quality Flags
+
+Add a flag for data quality, this is a numeric variable between 1 and 3 with 1 representing the best quality and 3 representing the worst. 
+
+First combine the subgroup data sets.
+```{r}
+metrics_income_sub <-results_all_expand %>% 
+  mutate(subgroup = "All",
+         subgroup_type = "all") %>% 
+  bind_rows(results_race_ethnicity_expanded)
+```
+
+First, join on to county_puma which provides the quality of the crosswalk (`afact * afact2`).
+```{r}
+
+metrics_education_sub <- left_join(metrics_income_sub, county_puma, by=c("crosswalk_period", "statefip","county"))
+```
+
+Now create the metric quality variable. The `share_in_preschool_quality` variable combines information on the quality of the crosswalk and the sample size (effective sample) to create a final quality flag. Note that any county with NA for the `share_in_preschool` metric is being suppressed due to sample size and is given a quality flag of 3.  
+
+```{r}
+metrics_education_sub <- metrics_education_sub %>% 
+         mutate(share_in_preschool_quality = case_when(is.na(share_in_preschool) ~ 3,
+                                                   geographic_allocation_quality == 1 ~ 1,
+                                                   geographic_allocation_quality == 2 ~ 2,
+                                                   geographic_allocation_quality == 3 ~ 3)
+  )
+```
+
+### Validation
```

Comment:
I think this could use some additional validation and exploration. Consider the examples below:

Additional validation:

```r
# check the missingness pattern -------------------------------------------
nonmissing <- metrics_education_all |>
  filter(!is.na(share_in_preschool) & !is.na(share_in_preschool_lb) & !is.na(share_in_preschool_ub))

missing <- metrics_education_all |>
  filter(is.na(share_in_preschool) & is.na(share_in_preschool_lb) & is.na(share_in_preschool_ub))

stopifnot(nrow(nonmissing) + nrow(missing) == nrow(metrics_education_all))


# check the order ---------------------------------------------------------
# there are three variables and we are interested in pairwise comparisons,
# therefore we need 3 choose 2 = 3 tests

# upper bound is always greater than the estimate
stopifnot(all(nonmissing$share_in_preschool_ub > nonmissing$share_in_preschool))

# upper bound is always greater than the lower bound
stopifnot(all(nonmissing$share_in_preschool_ub > nonmissing$share_in_preschool_lb))

# estimate is always greater than the lower bound
stopifnot(all(nonmissing$share_in_preschool > nonmissing$share_in_preschool_lb))

# visualize ---------------------------------------------------------------
metrics_education_all %>%
  ggplot(aes(share_in_preschool, share_in_preschool_lb)) +
  geom_point(alpha = 0.1) +
  labs(title = "The relationship between the estimate and the lower bound is sensible")

metrics_education_all %>%
  ggplot(aes(share_in_preschool_ub, share_in_preschool)) +
  geom_point(alpha = 0.1) +
  labs(title = "The relationship between the estimate and the upper bound is sensible")

metrics_education_all %>%
  ggplot(aes(share_in_preschool_ub, share_in_preschool_lb)) +
  geom_point(alpha = 0.1) +
  labs(title = "The relationship between the upper bound and the lower bound is sensible")


# distribution of confidence interval widths ------------------------------
metrics_education_all %>%
  mutate(ci_width = share_in_preschool_ub - share_in_preschool_lb) %>%
  pull(ci_width) %>%
  summary(na.rm = TRUE)



```

Reply:
I think you can also do some work exploring the change between 2018 and 2021. For example,

```r
metrics_education_all %>%
  group_by(geoid) %>%
  mutate(share_in_preschool_change = share_in_preschool - lag(share_in_preschool)) %>%
  filter(year == 2021) %>%
  ggplot(aes(share_in_preschool, share_in_preschool_change)) +
  geom_point(alpha = 0.2)
```

and

```r
metrics_education_all %>%
  select(year, geoid, share_in_preschool) %>%
  pivot_wider(
    names_from = year, 
    values_from = share_in_preschool
  ) %>%
  ggplot(aes(`2018`, `2021`)) +
  geom_abline() +
  geom_point(alpha = 0.1)
```